assets/Yves/js/modules/fact-finder-suggestions-box.js

```diff
@@ -15,15 +15,23 @@ var suggestionsBox = {
     prepareSuggestionsBlock: function (queryText, objectsList)     {
         this.clearProductSuggestionsBlock();
         this.clearCategoriesSuggestionsBlock();
+        this.clearSearchTermsSuggestionsBlock();
         this.resetCursorPosition();
 
         $.each(objectsList, function (i, item) {
-            if (item.type == 'productName') {
-                var productTemplateHtml = suggestionsBox.getProductTemplateHtml(item);
-                $('.ff-products').append(productTemplateHtml);
-            } else {
-                var categoryTemplateHtml = suggestionsBox.getCategoryTemplateHtml(item, queryText);
-                $('.ff-categories').append(categoryTemplateHtml);
+            switch (item.type) {
```

Comment:
Please check other possibilities how to implement this type-related logic in a more extendable way.